app/src/main/scala/org/alephium/explorer/service/MarketService.scala

```diff
@@ -212,12 +212,14 @@ object MarketService extends StrictLogging {
         assets: ArraySeq[TokenList.Entry],
         retried: Int
     ): Future[Either[String, ArraySeq[Price]]] = {
-      handleResponseAndRetryOnTooManyRequests(
+      handleResponseAndRetryWithCondition(
         "mobula/price",
         response,
+        _.code != StatusCode.Ok,
```

Comment:
Q: So in case of too many requests, they return 403 instead? 
Maybe we should not retry for 4xx error codes since those are usually non-transient. WDYT?

Reply:
yes they return 403, but it's actually transient, as if you retry few seconds latter it's all good.